ultralytics/nn/autobackend.py

```diff
@@ -199,7 +200,22 @@ def __init__(
                 device = torch.device("cpu")
                 cuda = False
             LOGGER.info(f"Preferring ONNX Runtime {providers[0]}")
-            session = onnxruntime.InferenceSession(w, providers=providers)
+            if imx:
+                check_requirements(
+                    ["model-compression-toolkit==2.1.1", "sony-custom-layers[torch]==0.2.0", "onnxruntime-extensions"]
+                )
+                w = [i for i in Path(w).iterdir() if i.name.endswith("onnx")][0]
```

Comment:
No, this is not the right way to find a file in a directory. See OpenVINO example that finds the *.xml file in the directory with glob.